File: go-sdk/sdk.go
package epilot

import (
	"net/http"

	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

var ServerList = []string{
	"https://user.sls.epilot.io",
}

type HTTPClient interface {
	Do(req *http.Request) (*http.Response, error)
}

type Epilot struct {
	UserV1 *UserV1
	UserV2 *UserV2

	_defaultClient  HTTPClient
	_securityClient HTTPClient
	_security       *shared.Security
	_serverURL      string
	_language       string
	_sdkVersion     string
	_genVersion     string
}

type SDKOption func(*Epilot)

func WithServerURL(serverURL string, params map[string]string) SDKOption {
	return func(sdk *Epilot) {
		if params != nil {
			serverURL = utils.ReplaceParameters(serverURL, params)
		}

		sdk._serverURL = serverURL
	}
}

func WithClient(client HTTPClient) SDKOption {
	return func(sdk *Epilot) {
		sdk._defaultClient = client
	}
}

func WithSecurity(security shared.Security) SDKOption {
	return func(sdk *Epilot) {
		sdk._security = &security
	}
}

func New(opts ...SDKOption) *Epilot {
	sdk := &Epilot{
		_language:   "go",
		_sdkVersion: "",
		_genVersion: "0.21.0",
	}
	for _, opt := range opts {
		opt(sdk)
	}

	if sdk._defaultClient == nil {
		sdk._defaultClient = http.DefaultClient
	}
	if sdk._securityClient == nil {

		if sdk._security != nil {
			sdk._securityClient = utils.ConfigureSecurityClient(sdk._defaultClient, sdk._security)
		} else {
			sdk._securityClient = sdk._defaultClient
		}

	}

	if sdk._serverURL == "" {
		sdk._serverURL = ServerList[0]
	}

	sdk.UserV1 = NewUserV1(
		sdk._defaultClient,
		sdk._securityClient,
		sdk._serverURL,
		sdk._language,
		sdk._sdkVersion,
		sdk._genVersion,
	)

	sdk.UserV2 = NewUserV2(
		sdk._defaultClient,
		sdk._securityClient,
		sdk._serverURL,
		sdk._language,
		sdk._sdkVersion,
		sdk._genVersion,
	)

	return sdk
}

File: user-api/java-sdk/lib/src/main/java/SDK.java


package ;

import .utils.HTTPClient;
import .utils.SpeakeasyHTTPClient;



public class Epilot {
	public static final String[] SERVERS = {
		"https://user.sls.epilot.io",
	};
  	
  	public UserV1 userV1;
  	public UserV2 userV2;	

	private HTTPClient _defaultClient;
	private HTTPClient _securityClient;
	private .models.shared.Security _security;
	private String _serverUrl;
	private String _language = "java";
	private String _sdkVersion = "";
	private String _genVersion = "0.21.0";

	public static class Builder {
		private HTTPClient client;
		private .models.shared.Security security;
		private String serverUrl;
		private java.util.Map<String, String> params = new java.util.HashMap<String, String>();

		private Builder() {
		}

		public Builder setClient(HTTPClient client) {
			this.client = client;
			return this;
		}
		
		public Builder setSecurity(.models.shared.Security security) {
			this.security = security;
			return this;
		}
		
		public Builder setServerURL(String serverUrl) {
			this.serverUrl = serverUrl;
			return this;
		}
		
		public Builder setServerURL(String serverUrl, java.util.Map<String, String> params) {
			this.serverUrl = serverUrl;
			this.params = params;
			return this;
		}
		
		public Epilot build() throws Exception {
			return new Epilot(this.client, this.security, this.serverUrl, this.params);
		}
	}

	public static Builder builder() {
		return new Builder();
	}

	private Epilot(HTTPClient client, .models.shared.Security security, String serverUrl, java.util.Map<String, String> params) throws Exception {
		this._defaultClient = client;
		
		if (this._defaultClient == null) {
			this._defaultClient = new SpeakeasyHTTPClient();
		}
		
		if (security != null) {
			this._security = security;
			this._securityClient = .utils.Utils.configureSecurityClient(this._defaultClient, this._security);
		}
		
		if (this._securityClient == null) {
			this._securityClient = this._defaultClient;
		}

		if (serverUrl != null && !serverUrl.isBlank()) {
			this._serverUrl = .utils.Utils.replaceParameters(serverUrl, params);
		}
		
		if (this._serverUrl == null) {
			this._serverUrl = SERVERS[0];
		}
		
		this.userV1 = new UserV1(
			this._defaultClient,
			this._securityClient,
			this._serverUrl,
			this._language,
			this._sdkVersion,
			this._genVersion
		);
		
		this.userV2 = new UserV2(
			this._defaultClient,
			this._securityClient,
			this._serverUrl,
			this._language,
			this._sdkVersion,
			this._genVersion
		);
	}
	
}